networkx/drawing/layout.py

```diff
@@ -1110,6 +1111,108 @@ def multipartite_layout(G, subset_key="subset", align="vertical", scale=1, cente
+    prevents  congestion of  highly connected  nodes due  to
+    strong forcing  between nodes.  Second, it  utilizes the
+    layout space  more effectively by preventing  large gaps
+    that  spring layout  tends  to create.  Lastly, the  arf
+    layout represents symmmetries in  the layout better than
+    the default spring layout.
+
+    Parameters
+    ----------
+    g : nx.Graph or nx.DiGraph
+        Networkx graph.
+    pos : dict
+        Initial  position of  the nodes.  If set  to None  a
+        random layout will be used using seed as input.
+    b : float
+        Controls the radius of  the circular space the graph
+        layout will be plotted in.
+    alpha : float
+        Attraction scalar between edges. Should be larger than 1.
+    etol : float
+        Termination tolerance
+    dt : float
+        Integration factor
+    max_iter : int
+        Max iterations before termination of the algorithm.
+    seed : int, RandomState instance or None  optional (default=None)
+        Set the random state for deterministic node layouts.
+        If int, `seed` is the seed used by the random number generator,
+        if numpy.random.RandomState instance, `seed` is the random
+        number generator,
+        if None, the random number generator is the RandomState instance used
+        by numpy.random.
+
+    References
+    - https://www.worldscientific.com/doi/abs/10.1142/S0129183107011558
+
+    Returns
+    -------
+    pos : dict
+        A dictionary of positions keyed by node.
+
+    Examples
+    --------
+    >>> G = nx.grid_graph((5, 5))
+    >>> pos = nx.arf_layout(G)
+
+    """
+    import numpy as np
+
+    assert alpha > 1, "Alpha should be larger than 1"
+    pos_tmp = nx.random_layout(g, seed=seed)
+    if pos is None:
+        pos = pos_tmp.copy()
+    else:
+        for node in g.nodes():
+            if node not in pos:
+                pos[node] = pos_tmp[node].copy()
+
+    # Initialize spring constant matrix
+    N = len(g)
+    # No nodes no computation
+    if N == 0:
+        return pos
+    K = np.zeros((N, N))
+    for idx, x in enumerate(g.nodes()):
+        for jdx, y in enumerate(g.nodes()):
+            if idx == jdx:
+                K[idx, jdx] = 0
+            elif g.has_edge(x, y) or g.has_edge(y, x):
+                K[idx, jdx] = alpha
+            else:
+                K[idx, jdx] = 1
+    # vectorize values
+    p = np.asarray(list(pos.values()))
+    bV = np.sqrt(N) * b  # scaling constant for circular space
+
+    # looping variables
+    error = etol + 1
+    n_iter = 0
+    while error > etol:
+        diff = p[:, None] - p[None]
+        A = np.linalg.norm(diff, axis=-1)[..., None]
+        # attraction_force - repulsions force
+        change = K[..., None] * diff - bV / A * diff
+        change = np.nansum(change, axis=0)
+        p += change * dt
+
+        error = np.linalg.norm(change, axis=-1).sum()
+        # error = abs(change).sum()
```

Comment:
I think `abs(change).sum()` is the same as `np.linalg.norm(change, ord=1)`. That is, it is the L_1 norm. But line 1208 is summing the norms. So, is the comment correct? Or am I just being confused by it.

Reply:
It is indeed the same, I forgot to take this comment out and opted for the norm to make it congruent with the math in the paper. I am in favor for the `norm` notation, but open to change it to 1209 with the proper axis added.